src/main/scala/viper/gobra/translator/implementations/components/OptionImpl.scala

```diff
@@ -104,7 +77,8 @@ class OptionImpl extends Options {
   /**
     * {{{
     * axiom optsome_over_get {
-    *   forall o : Option[T], e : T :: optget(o) == e <==> o == optsome(e)
+    *   forall o : Option[T], e : T :: { optget(o), optsome(e) }
+    *     (optget(o) == e) <==> (o == optsome(e))
```

Comment:
This axiom is unsound (incombination with other axioms). 
My - longer - explanation:
1) The axiom implies that get is injective without any restriction
2) If get is complete on type option, then get is not injective, because `get(none) == get(some(get(none)))` and `none != some(...)`

I also double checked with Martin and his - shorter - explanation is: substitute `o` with `none` and `e` with `get(none)`, then the axiom yields `none == some(get(none))` which again is invalid.

Reply:
Yes, that's indeed true; thanks for spotting that!

I updated the axioms to be similar to the ADT encoding example on http://viper.ethz.ch/examples/encoding-adts.html.

(I could make it slightly less elaborate if that's better, by just using a simple `tag` function to replace the constructor type business.)